pkg/virt-controller/services/template_test.go

```diff
@@ -868,14 +870,11 @@ var _ = Describe("Template", func() {
 			})
 		})
 		Context("with node selectors", func() {
-			It("should add node selectors to template", func() {
-				ovmfPath := ""
-				if svc.IsARM64() {
-					ovmfPath = "/usr/share/AAVMF"
-				} else {
-					ovmfPath = "/usr/share/OVMF"
-				}
+			table.DescribeTable("should add node selectors to template", func(arch string, ovmfPath string) {
 
+				testutils.UpdateFakeClusterConfig(configMapInformer, &kubev1.ConfigMap{
+					Data: map[string]string{virtconfig.OVMFPathKey: ovmfPath},
```

Comment:
I am not entirely sure how useful the test is like it is right now. I think we have a default for `arm64` which is different than for `amd64`. If we just change this value in the config here, then we are only testing that t he OVMF path is picked up correctly. I am not sure if we increase our test coverage with that in a good way. What do you think?

Reply:
> I am not entirely sure how useful the test is like it is right now. I think we have a default for `arm64` which is different than for `amd64`.

I am not sure what the "default" means. A total new configuration for different arch?

Reply:
By the way, I think there are many modules that detect the CPU arch by themselves like
```
pkg/virt-api/webhooks/mutating-webhook
pkg/virt-config/
pkg/virt-controller/services/
pkg/virt-launcher/virtwrap/
cmd/virt-handler/
```
It is not good if we have more CPU arch-related codes

Reply:
> It is not good if we have more CPU arch-related codes

Yes, definitely, that should disappear over time. For this test to be useful, I would expect that in `pkg/virt-config` we stop detecting the   architecture, since it still does that:

```
var DefaultOVMFPath = NewDefaultArch(runtime.GOARCH).GetDefaultOVMFPathForArch()
```

and instead allow injecting the architecture, to then get the defaults for the current architecture. Then the test would really test something. What do you think?

Reply:
> and instead allow injecting the architecture (or allow pre-creatin the whole default config for an architecture via a constructor), to then get the defaults for the current architecture. Then the test would really test something.

I agree with you. Where is a proper place to injecting the architecture or to construct the cpuarch based config?

Reply:
> I agree with you. Where is a proper place to injecting the architecture or to construct the cpuarch based config?
 How about injecting it in the `NewFakeClusterConfig*` functions? `ClusterConfig` needs something like an arch-config, and `GetConfig` in `CluserConfig` would make use of it.

Reply:
> > How about injecting it in the `NewFakeClusterConfig*` functions? `ClusterConfig` needs something like an arch-config, and `GetConfig` in `CluserConfig` would make use of it.

Sorry about replying to you late, I just come back to work from vacation. I will give it a try.